sklearn/utils/_metadata_requests.py

```diff
@@ -686,25 +686,26 @@ def __str__(self):
 # This section includes all objects required for MetadataRouter which is used
 # in routers, returned by their ``get_metadata_routing``.
 
-# This namedtuple is used to store a (mapping, routing) pair. Mapping is a
-# MethodMapping object, and routing is the output of `get_metadata_routing`.
-# MetadataRouter stores a collection of these namedtuples.
+# `RouterMappingPair` is used to store a (mapping, router) tuple where `mapping` is a
+# `MethodMapping` object and `router` is the output of `get_metadata_routing`.
+# `MetadataRouter` stores a collection of `RouterMappingPair` objects in its
+# `_route_mappings` attribute.
 RouterMappingPair = namedtuple("RouterMappingPair", ["mapping", "router"])
 
-# A namedtuple storing a single method route. A collection of these namedtuples
-# is stored in a MetadataRouter.
+# `MethodPair` is used to store a single method routing. `MethodMapping` stores a list
+# of `MethodPair` objects in its `_routes` attribute.
 MethodPair = namedtuple("MethodPair", ["caller", "callee"])
 
 
 class MethodMapping:
     """Stores the mapping between caller and callee methods for a router.
 
     This class is primarily used in a ``get_metadata_routing()`` of a router
-    object when defining the mapping between a sub-object (a sub-estimator or a
-    scorer) to the router's methods. It stores a collection of namedtuples.
```

Comment:
Not sure why you're removing the information here.

Reply:
I wanted to keep it first and precisize it, but then decided [to remove](https://github.com/scikit-learn/scikit-learn/pull/29489/commits/1315e275ad91d5ef8484811e2eb7eaabfe37f6ae) so we don't document private attributes in public API.

Why remove:
-  "It stores a collection of namedtuples." alone is not precise enough to be useful.
- Users don't need to know this implementation detail, because it doesn't make any difference for usage.
- I believe the sentence just below (saying that you can get the information by iterating over the object) is more useful, because it's exactly stating what is the information and what type is has.